loadtests/simulation.py

```diff
@@ -0,0 +1,318 @@
+
+    def _get_configuration(self):
+        # Loads is removing the extra information contained in the ini files,
+        # so we need to parse it again.
+        config_file = self.config['config']
+        # When copying the configuration files, we lose the config/ prefix so,
+        # try to read from this folder in case the file doesn't exist.
+        if not os.path.isfile(config_file):
+            config_file = os.path.basename(config_file)
+            if not os.path.isfile(config_file):
+                msg = 'Unable to locate the configuration file, aborting.'
+                raise LookupError(msg)
+        return Config(config_file).get_map('loads')
+
+    def api_url(self, path):
+        url = "{0}/v0/{1}".format(self.server_url.rstrip('/'), path)
+        return url
+
+    def collection_url(self, prefix=True):
+        if prefix:
+            return self.api_url(self.collection)
+        return '/' + self.collection
+
+    def record_url(self, record_id, prefix=True):
+        return self.collection_url(prefix) + '/%s' % record_id
+
+    def init_record(self, *args, **kwargs):
+        """Initialization that happens once per user.
+
+        :note:
+
+            This method is called as many times as number of users.
+        """
+        # Create at least some records for this user
+        max_initial_records = self.conf.get('max_initial_records', 100)
+        self.nb_initial_records = random.randint(3, max_initial_records)
+        self.batch_requests_size = self.conf.get('batch_requests_size', 25)
+
+    def setUp(self):
+        """Choose some random records in the whole collection.
+
+        :note:
+
+            This method is called as many times as number of hits.
+        """
+        resp = self.session.get(self.collection_url())
+        self.records = resp.json()['data']
+
+        # Create some records, if not any in collection.
+        if len(self.records) < self.nb_initial_records:
+            for i in range(self.nb_initial_records):
+                self.create()
+            resp = self.session.get(self.collection_url())
+            self.records = resp.json()['data']
+
+        # Pick a random record
+        self.random_record = random.choice(self.records)
+        self.random_id = self.random_record['id']
+        self.random_url = self.record_url(self.random_id)
+
+        # Pick another random, different
+        self.records.remove(self.random_record)
+        self.random_record_2 = random.choice(self.records)
+        self.random_id_2 = self.random_record_2['id']
+        self.random_url_2 = self.record_url(self.random_id_2)
+
+    def test_simulation(self):
+        """Choose a random action among available, if not frequent enough,
+        try again recursively.
+
+        :note:
+
+            This method is called as many times as number of hits.
+        """
+        action, percentage = random.choice(ACTIONS_FREQUENCIES)
+
+        forced_action = os.getenv('LOAD_ACTION')
```

Comment:
I don't think this will work when being run in a cluster (I don't think the env is passed along). Should we read from the configuration instead?